spinalcordtoolbox/straightening.py

```diff
@@ -389,12 +388,23 @@ def straighten(self):
                     lookup_curved2straight[index] = idx_closest
                 else:
                     lookup_curved2straight[index] = 0
-        for p in range(0, len(lookup_curved2straight) // 2):
```

Comment:
I see, this is why the bug only started showing up on lumbar images, where the halfway point isn't in the image, so this loop (among others) might stop too early.